script/validators/source_validator.py

```diff
@@ -127,19 +129,29 @@ def check_format(file_path):
         clang_format_cmd = [CLANG_FORMAT, "-style=file", file_path]
         formatted_src = subprocess.check_output(clang_format_cmd).splitlines(True)
         # Load source file
-        file = open(file_path, "r")
-        src = file.readlines()
+        with open(file_path, "r") as file:
+            src = file.readlines()
 
         # Do the diff
         diff = difflib.unified_diff(src, formatted_src)
-        for line in diff:
-            LOG.info("Invalid formatting in file : " + file_path)
+        if diff.next():  # Got a non-empty diff.
```

Comment:
The interator is incremented when you call `.next()`, so I think you miss the first element in the later loop. Or does the loop reset the iterator?